### PR TITLE
Add CurvedAnimation disposals in some widgets [prod-leak-fix]

### PR DESCRIPTION
**Pull request by ksokolovskyi**
Contributes to https://github.com/flutter/flutter/issues/141198

### Description
- Adds `CurvedAnimation` disposals to `material/chip.dart`, `material/input_decorator.dart`, `material/toggleable.dart`, `widgets/animated_switcher.dart`, `widgets/overscroll_indicator.dart`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I updated tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/overscroll_indicator.dart [hunk 1]
```diff
@@ -383,6 +384,8 @@ class _GlowController extends ChangeNotifier {
   @override
   void dispose() {
     _glowController.dispose();
+    _decelerator.removeListener(notifyListeners);
```
Comment (polina-c) [inline packages/flutter/lib/src/widgets/overscroll_indicator.dart:387]: Is it necessary to call `_decelerator.removeListener(notifyListeners);`?
Does not 'dispose' remove them?

If it is necessary, can you comment why?

**Review by polina-c — CHANGES_REQUESTED** [review]
Thank you! Looks good, but one line is confusing.

**Review by polina-c — APPROVED** [review]

File: packages/flutter/lib/src/material/toggleable.dart [hunk 2]
```diff
@@ -80,7 +80,7 @@ mixin ToggleableStateMixin<S extends StatefulWidget> on TickerProviderStateMixin
   /// To paint the actual radial reaction, [ToggleablePainter.paintRadialReaction]
   /// may be used.
   Animation<double> get reactionHoverFade => _reactionHoverFade;
```
Comment (Piinks) [inline packages/flutter/lib/src/material/toggleable.dart:82]: Should the getters also be updated with this type change?

File: packages/flutter/lib/src/material/toggleable.dart [hunk 3]
```diff
@@ -91,7 +91,7 @@ mixin ToggleableStateMixin<S extends StatefulWidget> on TickerProviderStateMixin
   /// To paint the actual radial reaction, [ToggleablePainter.paintRadialReaction]
   /// may be used.
   Animation<double> get reactionFocusFade => _reactionFocusFade;
```
Comment (Piinks) [inline packages/flutter/lib/src/material/toggleable.dart:93]: Same here

File: packages/flutter/lib/src/material/toggleable.dart [hunk 4]
```diff
@@ -69,7 +69,7 @@ mixin ToggleableStateMixin<S extends StatefulWidget> on TickerProviderStateMixin
   /// To paint the actual radial reaction, [ToggleablePainter.paintRadialReaction]
   /// may be used.
   Animation<double> get reaction => _reaction;
```
Comment (Piinks) [inline packages/flutter/lib/src/material/toggleable.dart:71]: And here

File: packages/flutter/lib/src/material/toggleable.dart (hunk 2 above)
Comment (ksokolovskyi) [reply packages/flutter/lib/src/material/toggleable.dart]: @Piinks, thanks for review. I've replaced `Animation<double>` with `CurvedAnimation` in all the places you mentioned.

**Review by Piinks — APPROVED** [review]
LGTM

**Comment by ksokolovskyi** [comment]
@polina-c @Piinks thanks for the review!

**Pull request merged**